eng/pipelines/common/templates/runtimes/run-test-job.yml

```diff
@@ -62,6 +62,9 @@ jobs:
       - ${{ if ne(parameters.liveLibrariesBuildConfig, '') }}:
         - ${{ format('libraries_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.liveLibrariesBuildConfig) }}
 
+    - ${{ if eq(parameters.runtimeFlavor, 'mono') }}:
+      - ${{ format('mono_product_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, coalesce(parameters.liveRuntimeBuildConfig, parameters.buildConfig)) }}
```

Comment:
this is because `${{ if ne(parameters.stagedBuild, true) }}` is false right? cause otherwise it should've already got the dependency on L61 above

Reply:
RIght, actually stagedBuild is never set, this is a leftover from coreclr's build templates. We shouldn't need this change. Good catch @akoeplinger

Reply:
We transitively depend on the  coreclr product, because the common_test_build on line 57 depends on it.

Reply:
```
- ${{ format('{0}_product_build_{1}{2}_{3}_{4}', parameters.runtimeFlavor, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
```

`runtimeFlavor` is mono in this case, so it will translate to mono. I don't think you need to add this. Since `build-test-job` already depends on coreclr and mono product build and this job depends on `build-test-job` this job already depends on the right things transitively.

Reply:
hm something didn't work cause we saw `##[error]Artifact MonoProduct_OSX_x64_release not found for build 611688. Please ensure you have published artifacts in any previous phases of the current build.` in Attempt 1 of the PR build https://dev.azure.com/dnceng/public/_build/results?buildId=611688&view=logs&j=d3d95fb7-3682-5a91-e55a-5fa7fd764a86&t=085fb874-fb75-5ef7-1ed5-64f6fb66e939 which shouldn't be possible if the dependency is correct.

Reply:
What is missing is the dependency to mono's product build in `build-test-job` not in `run-test-job`. From the expanded templates these are the dependencies of `build-test-job`:

```
- job: mono_common_test_build_p0_OSX_x64_release
    condition: >-
      and(succeeded(), and(succeeded(), or(
        eq(dependencies.checkout.outputs['SetPathVars_mono.containsChange'], true),
        eq(variables['isFullMatrix'], true))))
    continueOnError: false
    dependsOn:
    - checkout
    - coreclr_product_build_OSX_x64_release
    - libraries_build_OSX_x64_Debug
    displayName: Mono Common Pri0 Test Build OSX x64 release
```

These are the dependencies of `run-test-job`:
```
- job: run_test_p0_mono__OSX_x64_release
    condition: >-
      and(succeeded(), or(
        eq(dependencies.checkout.outputs['SetPathVars_mono.containsChange'], true),
        eq(variables['isFullMatrix'], true)))
    continueOnError: false
    dependsOn:
    - checkout
    - mono_common_test_build_p0_OSX_x64_release
    - mono_product_build_OSX_x64_release
    - libraries_build_OSX_x64_Debug
    displayName: Mono  Pri0 Test Run  OSX x64 release
```

https://dev.azure.com/dnceng/public/_apis/build/builds/611688/logs?logid=1

Reply:
build-test-job doesn't depend on mono product build though; we don't actually need the mono product build to do the test build (just coreclr). 

So if stagedBuild is false, we have no dependency on mono product build, transitive or otherwise.  It looks like stagedBuild is usually false; I'm not exactly sure under what circumstances it is true.

Reply:
Actually, I don't think we need the mono product build until we are actually do the runtime patching step in runtest. So I think the error is that we download mono-product in build-test-job in the first place.